### ISSUE TITLE
The agent on Windows has crashed

### ISSUE THREAD

**Issue by vikman90** [wazuh/wazuh#7836]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.1.2 | Agent service | Agent | Packages | Windows 8 |

## Crash report

<details>
  <summary>Report</summary>

```
Version=1
EventType=APPCRASH
EventTime=132598629568993331
ReportType=2
Consent=1
ReportIdentifier=82ca7860-81b3-11eb-b827-005056962179
IntegratorReportIdentifier=5df19b18-eac0-41ca-b40d-b40c83613a4f
WOW64=1
NsAppName=ossec-agent.exe
AppSessionGuid=000012f8-0000-002e-645c-463ebd15d701
TargetAppId=W:0000ba5cf28d3c7f3b889b0e166a9921ad7c00000904!000085adce91ca7c894ef9868557cd22e15deb2a93be!ossec-agent.exe
TargetAppVer=2021//03//05:13:47:00!1d8bac!ossec-agent.exe
BootId=4294967295
Response.type=4
Sig[0].Name=Application Name
Sig[0].Value=ossec-agent.exe
Sig[1].Name=Application Version
Sig[1].Value=0.0.0.0
Sig[2].Name=Application Timestamp
Sig[2].Value=60423654
Sig[3].Name=Fault Module Name
Sig[3].Value=ossec-agent.exe
Sig[4].Name=Fault Module Version
Sig[4].Value=0.0.0.0
Sig[5].Name=Fault Module Timestamp
Sig[5].Value=60423654
Sig[6].Name=Exception Code
Sig[6].Value=c0000005
Sig[7].Name=Exception Offset
Sig[7].Value=0000e6e5
DynamicSig[1].Name=OS Version
DynamicSig[1].Value=10.0.14393.2.0.0.272.7
DynamicSig[2].Name=Locale ID
DynamicSig[2].Value=1033
DynamicSig[22].Name=Additional Information 1
DynamicSig[22].Value=6582
DynamicSig[23].Name=Additional Information 2
DynamicSig[23].Value=65827c5ac1cffbf1d2cbfbed3c201de5
DynamicSig[24].Name=Additional Information 3
DynamicSig[24].Value=c34d
DynamicSig[25].Name=Additional Information 4
DynamicSig[25].Value=c34d1c9a497c5d7acf9652a3d0565a36
UI[2]=C:\Program Files (x86)\ossec-agent\ossec-agent.exe
UI[5]=Check online for a solution (recommended)
UI[6]=Check for a solution later (recommended)
UI[7]=Close
UI[8]=ossec-agent.exe stopped working and was closed
UI[9]=A problem caused the application to stop working correctly. Windows will notify you if a solution is available.
UI[10]=&Close
LoadedModule[0]=C:\Program Files (x86)\ossec-agent\ossec-agent.exe
LoadedModule[1]=C:\Windows\SYSTEM32\ntd1l.dll
LoadedModule[2]=C:\Windows\System32\kern3l32.dll
LoadedModule[3]=C:\Windows\SYSTEM32\ntdll.dll
LoadedModule[4]=C:\Windows\System32\KERNEL32.DLL
LoadedModule[5]=C:\Windows\System32\KERNELBASE.dll
LoadedModule[6]=C:\Program Files\SentinelOne\Sentinel Agent 4.2.5.168\InProcessClient32.dll
LoadedModule[7]=C:\Windows\System32\ADVAPI32.dll
LoadedModule[8]=C:\Windows\System32\msvcrt.dll
LoadedModule[9]=C:\Windows\System32\sechost.dll
LoadedModule[10]=C:\Windows\System32\RPCRT4.dll
LoadedModule[11]=C:\Windows\System32\SspiCli.dll
LoadedModule[12]=C:\Windows\System32\CRYPTBASE.dll
LoadedModule[13]=C:\Windows\System32\bcryptPrimitives.dll
LoadedModule[14]=C:\Windows\System32\PSAPI.DLL
LoadedModule[15]=C:\Windows\System32\SHLWAPI.dll
LoadedModule[16]=C:\Windows\System32\combase.dll
LoadedModule[17]=C:\Program Files (x86)\ossec-agent\libwinpthread-1.dll
LoadedModule[18]=C:\Windows\System32\ucrtbase.dll
LoadedModule[19]=C:\Windows\SYSTEM32\IPHLPAPI.DLL
LoadedModule[20]=C:\Windows\System32\GDI32.dll
LoadedModule[21]=C:\Windows\System32\gdi32full.dll
LoadedModule[22]=C:\Windows\System32\USER32.dll
LoadedModule[23]=C:\Windows\System32\win32u.dll
LoadedModule[24]=C:\Windows\System32\WS2_32.dll
LoadedModule[25]=C:\Windows\SYSTEM32\wevtapi.dll
LoadedModule[26]=C:\Windows\SYSTEM32\WSOCK32.DLL
LoadedModule[27]=C:\Program Files (x86)\ossec-agent\libwazuhext.dll
LoadedModule[28]=C:\Windows\SYSTEM32\bcrypt.dll
LoadedModule[29]=C:\Windows\System32\CRYPT32.dll
LoadedModule[30]=C:\Windows\System32\MSASN1.dll
LoadedModule[31]=C:\Windows\SYSTEM32\VERSION.dll
LoadedModule[32]=C:\Windows\SYSTEM32\CRYPTSP.dll
LoadedModule[33]=C:\Windows\system32\rsaenh.dll
LoadedModule[34]=C:\Windows\System32\profapi.dll
LoadedModule[35]=C:\Windows\system32\mswsock.dll
LoadedModule[36]=C:\Windows\System32\NSI.dll
LoadedModule[37]=C:\Windows\SYSTEM32\dhcpcsvc6.DLL
LoadedModule[38]=C:\Windows\SYSTEM32\dhcpcsvc.DLL
LoadedModule[39]=C:\Program Files (x86)\ossec-agent\syscollector_win_ext.dll
LoadedModule[40]=C:\Windows\system32\napinsp.dll
LoadedModule[41]=C:\Windows\SYSTEM32\DNSAPI.dll
LoadedModule[42]=C:\Windows\System32\winrnr.dll
LoadedModule[43]=C:\Windows\system32\NLAapi.dll
LoadedModule[44]=C:\Windows\SYSTEM32\ntmarta.dll
FriendlyEventName=Stopped working
ConsentKey=APPCRASH
AppName=ossec-agent.exe
AppPath=C:\Program Files (x86)\ossec-agent\ossec-agent.exe
NsPartner=windows
NsGroup=windows8
ApplicationIdentity=62C8EE9F91BFFB63B51E5EE4E680E67A
MetadataHash=1351108878
```

</details>

## Log

<details>
  <summary>ossec.log</summary>

```
2021/03/10 09:54:17 ossec-agent: INFO: (6000): Starting daemon...
2021/03/10 09:54:17 rootcheck: INFO: Starting rootcheck scan.
2021/03/10 09:54:17 ossec-agent: INFO: (6010): File integrity monitoring scan frequency: 43200 seconds
2021/03/10 09:54:17 ossec-agent: INFO: (6008): File integrity monitoring scan started.
2021/03/10 09:54:17 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2021/03/10 09:54:20 sca: INFO: Evaluation finished for policy 'C:\Program Files (x86)\ossec-agent\ruleset\sca\sca_win_audit.yml'
2021/03/10 09:54:20 sca: INFO: Security Configuration Assessment scan finished. Duration: 4 seconds.
2021/03/10 09:54:20 wazuh-modulesd:syscollector: INFO: Evaluation finished.
2021/03/10 09:54:22 rootcheck: INFO: Ending rootcheck scan.
2021/03/10 09:54:38 ossec-agent: INFO: (6009): File integrity monitoring scan ended.
2021/03/10 09:54:38 ossec-agent: INFO: (6012): Real-time file integrity monitoring started.
```

</details>

**Comment by hanschl** [comment]
|Wazuh Client version|Component|Wazuh Server version|Install type|Install method|Platform Client|Platform Server|
|---|---|---|---|---|---|---|
| 4.1.2 | Wazuh Agent Windows| 4.1.2 | Agent | Packages|See Description|CentOS 8|

Hello,

we're experiencing the same problem on our Windows servers. As soon as we updated the client to 4.1.2.1 Wazuh started crashing on some of our Windows servers.

Affected are the following OS:
- Windows Server 2019
- Windows Server 2016
- Windows Server 2012

If you need any more information please let me know.

Best Regards

**Comment by hanschl** [comment]
Hello,

I did some testing. The last version that is working for us is version 4.0.4. Everything after that is crashing. 

I've activated the debug log but there ist not much I can see. There is an Error that the agent can't extract the permissions of pagefile but that is normal cause Windows doesn't set any for pagefile and hiberfile.

Syscheck is done and after that the agent is running for about 10 minutes before crashing without writing any errors to log:

# ossec debug log
<details>

```
2021/03/16 10:30:26 ossec-agent[5884] create_db.c:1206 at fim_print_info(): DEBUG: (6330): The scan has been running during: 865.656 sec (865.672 clock sec)
2021/03/16 10:30:26 ossec-agent[5884] create_db.c:1209 at fim_print_info(): DEBUG: (6335): Fim entries: 38512
2021/03/16 10:30:27 ossec-agent[5884] create_db.c:1210 at fim_print_info(): DEBUG: (6354): Fim registry entries: 36906
2021/03/16 10:30:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:30:33 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:30:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:30:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:31:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:31:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:31:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:31:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:31:38 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:31:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:31:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:32:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:32:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:32:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:32:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:32:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:32:43 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:32:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:33:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:33:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:33:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:33:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:33:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:33:48 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:33:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:34:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:34:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:34:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:34:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:34:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:34:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:34:53 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:35:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:35:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:35:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:35:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:35:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:35:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:35:58 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:36:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:36:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:36:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:36:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:36:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:36:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:37:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:37:03 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:37:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:37:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:37:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:37:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:37:45 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:37:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:37:52 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:38:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:38:08 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:38:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:38:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:38:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:38:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:38:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:39:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:39:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:39:13 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:39:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:39:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:39:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:39:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:40:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:40:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:40:17 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:40:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:40:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:40:39 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:40:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:40:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:40:51 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:40:53 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:40:59 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:41:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:41:02 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:41:04 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:41:09 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:41:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:41:11 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:41:17 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:41:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:41:20 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:41:22 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:41:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:41:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:41:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:41:53 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:42:00 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:42:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:42:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:42:20 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:42:23 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:42:27 ossec-agent[5884] logcollector.c:346 at LogCollectorStart(): DEBUG: Performing file check.
2021/03/16 10:42:28 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:42:30 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:42:38 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:42:40 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:42:46 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
2021/03/16 10:42:50 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:43:00 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:43:10 ossec-agent[5884] notify.c:115 at run_notify(): DEBUG: Sending agent notification.
2021/03/16 10:43:14 ossec-agent[5884] fim_db.c:380 at fim_db_check_transaction(): DEBUG: Database transaction completed.
```
</details>

# Windows App Log
<details>

```
Log Name:      Application
Source:        Application Error
Date:          16.03.2021 10:43:14
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      
Description:
Faulting application name: ossec-agent.exe, version: 0.0.0.0, time stamp: 0x60127559
Faulting module name: ossec-agent.exe, version: 0.0.0.0, time stamp: 0x60127559
Exception code: 0xc0000005
Fault offset: 0x0000e6e5
Faulting process id: 0x16fc
Faulting application start time: 0x01d71a416a7ef501
Faulting application path: C:\Program Files (x86)\ossec-agent\ossec-agent.exe
Faulting module path: C:\Program Files (x86)\ossec-agent\ossec-agent.exe
Report Id: f6043ec0-6089-4272-9018-6646ab679ce4
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2021-03-16T09:43:14.407615700Z" />
    <EventRecordID>16639</EventRecordID>
    <Channel>Application</Channel>
    <Computer></Computer>
    <Security />
  </System>
  <EventData>
    <Data>ossec-agent.exe</Data>
    <Data>0.0.0.0</Data>
    <Data>60127559</Data>
    <Data>ossec-agent.exe</Data>
    <Data>0.0.0.0</Data>
    <Data>60127559</Data>
    <Data>c0000005</Data>
    <Data>0000e6e5</Data>
    <Data>16fc</Data>
    <Data>01d71a416a7ef501</Data>
    <Data>C:\Program Files (x86)\ossec-agent\ossec-agent.exe</Data>
    <Data>C:\Program Files (x86)\ossec-agent\ossec-agent.exe</Data>
    <Data>f6043ec0-6089-4272-9018-6646ab679ce4</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>
```
</details>

**Comment by jordanenglish** [comment]
@hanschl - same. We've had to revert to 4.0.4

**Comment by vikman90** [comment]
Hi @jordanenglish,

Still working on this. That information is very helpful to narrow down the cause of the bug.

Will keep you posted. Thank you for notifying.

**Comment by antoniomanuelfr** [comment]
Hi all. 
Other users had the same problem (see #8013) and shared some logs that proofed helpful in narrowing down the problem.
The problematic code is the following: https://github.com/wazuh/wazuh/blob/bd86b194d6af7ee3daf60209aa40c4c09ef10ba7/src/syscheckd/db/fim_db.c#L301-L344
We use the character `:` to delimit where a registry key ends and the registry value begins in synchronization messages. When a value inside a monitored key starts with `:` we have the following sync message:
```
2021/03/29 14:17:40 ossec-agent[2208] run_check.c:78 at fim_send_sync_msg(): DEBUG: (6317): Sending integrity control message: {"component":"fim_registry","type":"integrity_check_global","data":{"id":1617027460,"begin":"[x64] HKEY_LOCAL_MACHINE\\Software\\test:","end":"[x64] HKEY_LOCAL_MACHINE\\Software\\test\\subkey:::test_value","checksum":"007d8363f1e8fb7cf219fc38a5ee8ab9daa9caef"}}
``` 
When the function finds ` HKEY_LOCAL_MACHINE\\Software\\test\\subkey:::`, the last `:` is replaced by a null character, and the `::` are replaced by one `:`. The final path of the registry key that will be checked is `HKEY_LOCAL_MACHINE\\Software\\test\\subkey:` (which doesn't exist) the code fails to check if the function `fim_db_get_registry_key` returns `NULL`, and later, we have a null dereference that causes the crash on the windows agent. 

We used this approach because every single character can be used in registries, and this corner case passed our tests. 

I would like to add that we have already created new tests that will check this behaviour and prevent this kind of bugs in the future.

We will fix this as soon as possible. 

Kind regards

**Comment by c0r3dump3d** [comment]
The bug it's corrected? I have tested the version wazuh-agent-4.1.4-1.msi but the bugs it's still present ...

**Comment by antoniomanuelfr** [comment]
Hi @c0r3dump3d.
The crash is fixed in the 4.2 branch, but some tweaking in the synchronization mechanism must be done in order to handle the problematic values. We are currently working on this and hope that it will be ready ASAP. 

In the meantime, you have two options: 
- Disable the registry synchronization mechanism: 
```xml
    <synchronization>
      <enabled>yes</enabled>
      <interval>5m</interval>
      <max_interval>1h</max_interval>
      <max_eps>10</max_eps>
      <registry_enabled>no</registry_enabled>
    </synchronization>
  </syscheck>
```
With this option, you prevent the synchronization of the registries between the agent and the manager and preventing the crash.

- The other option is to not monitor the problematic key. To find out the key, you can enable the syscheck debug mode and the agent will crash printing this message: 
```
2021/04/07 09:21:39 ossec-agent[2444] receiver-win.c:128 at receiver_thread(): DEBUG: Received message: '#!-fim_registry dbsync no_data {"id":1617787299,"begin":"[x64] HKEY_LOCAL_MACHINE\\Software\\test:::testvalue","end":"[x64] HKEY_LOCAL_MACHINE\\Software\\test:::testvalue"}'
```
The problematic key is `[x64] HKEY_LOCAL_MACHINE\\Software\\test`. So if you remove this key from the monitoring, the agent won't crash. 

I know this is not the best solution, but as I mentioned earlier, we are working to fix this ASAP.

**Comment by c0r3dump3d** [comment]
Thank's for the answer, for now, until the update, I wiil disable the registry synchronization ...
